packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -232,6 +258,43 @@ function WPNonProductLineItem( {
 	/* eslint-enable wpcalypso/jsx-classname-namespace */
 }
 
+function WPCouponLineItem( {
+	lineItem,
+	className,
+	isSummary,
+	hasDeleteButton,
+	removeProductFromCart,
+	createUserAndSiteBeforeTransaction,
+	isPwpoUser,
+	hasPartnerCoupon,
+}: {
+	lineItem: LineItemType;
+	className?: string | null;
+	isSummary?: boolean;
+	hasDeleteButton?: boolean;
+	removeProductFromCart?: () => void;
+	createUserAndSiteBeforeTransaction?: boolean;
+	isPwpoUser?: boolean;
+	hasPartnerCoupon?: boolean;
+} ): JSX.Element {
+	return (
+		<div
+			className={ joinClasses( [ className, 'coupon-line-item' ] ) }
+			data-partner-coupon={ !! hasPartnerCoupon }
```

Comment:
_(More of a question than anything - it wouldn't change the final outcome):_ Do we use this solely to render the component differently? If yes, how come we went with an attribute instead of a class here? 😄